src/current/v23.1/cdc-queries.md

```diff
@@ -65,7 +65,7 @@ Function                  | Description
 --------------------------+----------------------
 `changefeed_creation_timestamp()` | Returns the decimal MVCC timestamp when the changefeed was created.
 `event_op()`              | Returns a string describing the type of event. If a changefeed is running with the [`diff`]({% link {{ page.version.version }}/create-changefeed.md %}#diff-opt) option, then this function returns `'insert'`, `'update'`, or `'delete'`. If a changefeed is running without the `diff` option, it is not possible to determine an update from an insert, so `event_op()` returns [`'upsert'`](https://www.cockroachlabs.com/blog/sql-upsert/) or `'delete'`.
-`event_schema_timestamp()` | Returns the timestamp of the [schema changes]({% link {{ page.version.version }}/online-schema-changes.md %}) event.
+`event_schema_timestamp()` | Returns the timestamp of [schema change]({% link {{ page.version.version }}/online-schema-changes.md %}) events that cause a [changefeed message]({% link {{ page.version.version }}/changefeed-messages.md %}) to emit. When the schema change event does not result in a table backfill or scan, `event_schema_timestamp()` will return the event's timestamp. When the schema change event does result in a table backfill or scan, `event_schema_timestamp()` will return the timestamp at which the backfill/scan is read — the [high-water mark time]({% link {{ page.version.version }}/how-does-an-enterprise-changefeed-work.md %}) of the changefeed.
```

Comment:
that "how does an enterprise changefeed work" doc didnt have any info about hwms in there. or did i miss it?

otherwise i think this is a good high level summary of the function.

my reading of the code gives the following logic:

```
given highWater = the largest timestamp at or below which we know all events in the watched span have been seen 

if backfill:
  if initial scan:
    schemats = high water
  else:
    schemats = high water . next() (the ts immediately after hwm)
else:
  schemats = event ts
```

Reply:
The link summarizes within the body of the description: `... to update the high-water mark timestamp. The high-water mark acts as a checkpoint for the changefeed’s job progress, and guarantees that all changes before (or at) the timestamp have been emitted.` It is the best place for a "explanation" of the hwm. I want to work on terminology definitions throughout the cdc docs, so this is a good point. I will leave in there for now, but keep in mind that this should be updated.

Reply:
Thanks, your explanation here @asg0451 was very helpful — I appreciate it!